src/ocf_core.c

```diff
@@ -181,6 +181,9 @@ static inline int ocf_core_validate_io(struct ocf_io *io)
 	if (io->addr + io->bytes > ocf_volume_get_length(volume))
 		return -OCF_ERR_INVAL;
 
+	if (io->bytes == 0)
+		return -OCF_ERR_INVAL;
+
```

Comment:
I think it's expected to receive e.g. write request with size == 0 - it would make sense for example if the I/O has FLUSH flag set. Let's analyze this more carefully before we introduce this change.